src/main/java/com/aws/greengrass/crypto/CryptoI.java

```diff
@@ -0,0 +1,138 @@
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.List;
+import javax.inject.Inject;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509KeyManager;
+import javax.net.ssl.X509TrustManager;
+import javax.security.auth.x500.X500Principal;
+
+public interface CryptoI extends AutoCloseable {
+    AwsIotMqttConnectionBuilder getIotMqttConnectionBuilder();
+
+    X509KeyManager[] getKeyManagers() throws TLSAuthException;
+
+    X509TrustManager[] getTrustManagers() throws TLSAuthException;
+
+    void setImplementationUpdated(Runnable implChanged);
+
+    class Default implements CryptoI {
+        protected final String certificatePath;
+        protected final String privateKeyPath;
+        protected final String rootCAPath;
+        protected X509KeyManager[] keyManagers = null;
+        protected X509TrustManager[] trustManagers = null;
+        protected Runnable implChanged;
+        @SuppressWarnings("PMD.NullAssignment")
+        protected final Subscriber subscriber = (what, topic) -> {
+            if (!WhatHappened.timestampUpdated.equals(what) && !WhatHappened.initialized.equals(what)
+                    && implChanged != null) {
+                synchronized (this) {
+                    keyManagers = null;
+                    trustManagers = null;
+                }
+                implChanged.run();
+            }
+        };
+
+        @Inject
+        public Default(DeviceConfiguration deviceConfiguration) {
+            certificatePath = Coerce.toString(deviceConfiguration.getCertificateFilePath().subscribe(subscriber));
+            privateKeyPath = Coerce.toString(deviceConfiguration.getPrivateKeyFilePath().subscribe(subscriber));
+            rootCAPath = Coerce.toString(deviceConfiguration.getRootCAFilePath().subscribe(subscriber));
+        }
+
+        @Override
+        public AwsIotMqttConnectionBuilder getIotMqttConnectionBuilder() {
+            return AwsIotMqttConnectionBuilder.newMtlsBuilderFromPath(certificatePath, privateKeyPath)
+                    .withCertificateAuthorityFromPath(null, rootCAPath);
+        }
+
+        @Override
+        public synchronized X509KeyManager[] getKeyManagers() throws TLSAuthException {
+            if (keyManagers == null) {
+                keyManagers = createKeyManagers(privateKeyPath, certificatePath);
+            }
+            return keyManagers;
+        }
+
+        @Override
+        public synchronized X509TrustManager[] getTrustManagers() throws TLSAuthException {
+            if (trustManagers == null) {
+                trustManagers = createTrustManagers(rootCAPath);
+            }
+            return trustManagers;
+        }
+
+        @Override
+        public void setImplementationUpdated(Runnable implChanged) {
```

Comment:
What's the use case for this method?

Reply:
Allows the implementation to tell the CryptoProvider that something changed in order to trigger the onChange callbacks. Such as the certificate paths updating.